File: an/Autocomplete/app/src/main/java/com/example/autocompletedemo/MainActivity.java
package com.example.autocompletedemo;

import androidx.annotation.RequiresApi;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.view.AccessibilityDelegateCompat;
import androidx.core.view.ViewCompat;
import androidx.core.view.accessibility.AccessibilityNodeInfoCompat;

import android.annotation.SuppressLint;
import android.os.Build;
import android.os.Bundle;
import android.view.View;
import android.widget.ArrayAdapter;
import android.widget.AutoCompleteTextView;

public class MainActivity extends AppCompatActivity {
    AutoCompleteTextView autocomplete, autocomplete2;

    String[] arr = { "Paries,France", "PA,United States","Parana,Brazil",
            "Padua,Italy", "Pasadena,CA,United States"};
    String[] cou = {"China", "Germany", "Netherlands", "Portugla", "France", "Spain"};

    @SuppressLint("WrongViewCast")
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        autocomplete = (AutoCompleteTextView) findViewById(R.id.autoCompleteTextView1);

        ArrayAdapter<String> adapter = new ArrayAdapter<String>
                (this,android.R.layout.select_dialog_item, arr);

        autocomplete.setThreshold(2);
        autocomplete.setAdapter(adapter);

        autocomplete2 = (AutoCompleteTextView) findViewById(R.id.autoCompleteTextView2);
        ArrayAdapter<String> adapter2 = new ArrayAdapter<String>
                (this,android.R.layout.select_dialog_item, cou);


        autocomplete2.setThreshold(2);
        autocomplete2.setAdapter(adapter2);

        autocomplete.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (autocomplete.isPopupShowing()) {
                    autocomplete2.setImportantForAccessibility(View.IMPORTANT_FOR_ACCESSIBILITY_NO_HIDE_DESCENDANTS);
                }
            }
        });
    }


}

File: an/Wechat/app/src/main/java/com/example/wechat/MainActivity.kt
package com.example.wechat

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.ImageView

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val imageView1 = findViewById<ImageView>(R.id.image1)
        val imageView2 = findViewById<ImageView>(R.id.image2)
        val imageView3 = findViewById<ImageView>(R.id.image3)
        val imageView4 = findViewById<ImageView>(R.id.image4)

        imageView1.setOnClickListener(listener)
        imageView2.setOnClickListener(listener)
        imageView3.setOnClickListener(listener)
        imageView4.setOnClickListener(listener)
    }

    val listener = View.OnClickListener {view ->

        val trans = supportFragmentManager.beginTransaction()
        when(view.id) {
            R.id.image1 ->  trans.replace(R.id.container, Wechat_Fragment())
            R.id.image2 ->  trans.replace(R.id.container, Message_Fragment())
            R.id.image3 ->  trans.replace(R.id.container, Find_Fragment())
            else ->  trans.replace(R.id.container, Me_Fragment())
        }
        trans.commit()
    }
}
